org.idempiere.webservices/WEB-INF/src/org/idempiere/adinterface/Process.java

```diff
@@ -330,7 +330,7 @@ else if (AD_Menu_ID > 0 && AD_Process_ID <= 0)
 			{
 				int AD_Workflow_ID = process.getAD_Workflow_ID();
 				MWorkflow wf = MWorkflow.get (Env.getCtx(), AD_Workflow_ID);
-				MWFProcess wfProcess = wf.startWait(pi);	//	may return null
+				MWFProcess wfProcess = wf.start(pi, (String)null);	//	may return null

```

Comment:
Hi @hengsin, 

This code change does not look like a code readability improvement but a functionality change.

Reply:
I've deprecated MWorkflow.startWait (workflow process is synchronous so there's nothing to wait for here) and this change is needed to avoid deprecation warning.
If that's consider potentially a functionality change, I can revert both changes.

Reply:
Got it - looks good

Reply:
Hi @hengsin - while reviewing this line I think that's a mistake in processing a workflow in SOAP webservices.

It is not caused by this change, but it comes like that since always.

In my opinion, processing the workflow out of transaction sounds wrong, that line must be:
```java
    MWFProcess wfProcess = wf.start(pi, trxName);	//	may return null
```

or start a new transaction like in line 358, but processing without transaction sounds dangerous.

But that would be a matter of a different ticket because is not a readability issue, but more like a bug fix.

WDYT?

Reply:
hi @CarlosRuiz-globalqss , yes, that sounds like a valid concern. Please open a new ticket to fix that.